packages/api/cms-api/src/user-management/current-user.ts

```diff
@@ -0,0 +1,93 @@
+import { Field, ObjectType } from "@nestjs/graphql";
+import { GraphQLJSONObject } from "graphql-type-json";
+
+import { ContentScope } from "../common/decorators/content-scope.interface";
+import { PermissionConfiguration } from "./user-management.types";
+
+interface CurrentUserInterface {
+    permissions: CurrentUserPermission[];
+    contentScopes: CurrentUserContentScope[];
+    isAllowed(permission: UserPermission, contentScope: ContentScope): boolean;
+}
+
+@ObjectType()
+export class CurrentUserPermission {
+    @Field()
+    name: string;
+    @Field(() => GraphQLJSONObject, { nullable: true })
+    configuration?: PermissionConfiguration;
+    @Field()
+    overrideContentScopes: boolean;
```

Comment:
Do I understand this correctly:

- when overrideContentScopes=true the CurrentUserPermission.contentScopes are used
- when overrideContentScopes=false the CurrentUser.contentScopes are used